### PR TITLE
Added image publisher node.

### PR DESCRIPTION
**Pull request by dcrossman**
This reads from shared memory and publishes
the left and right camera images.
Also added the SharedImageReader and SharedImageWriter
libraries though the writer is currently unused.

### CODE REVIEW

**Comment by dcrossman** [comment]
We'll also need to add the
ros-indigo-image-transport 
and probably the 
ros-indigo-image-transport-plugins
packages to the robosub build.

**Comment by irwineffect** [comment]
I added those two packages to our robosub package!

File: src/vision/image_publisher.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    ros::Rate loop_rate(5);
+
+    while (ros::ok())
+    {
+        cv::Mat l = image_reader_left.Read();
+        cv::Mat r = image_reader_right.Read();
+
+        sensor_msgs::ImagePtr l_msg = cv_bridge::CvImage(std_msgs::Header(), "bgr8", l).toImageMsg();
+        sensor_msgs::ImagePtr r_msg = cv_bridge::CvImage(std_msgs::Header(), "bgr8", r).toImageMsg();
```
Comment (irwineffect) [inline src/vision/image_publisher.cpp:30]: Can we rename the topics to camera/left/image_raw and camera/right/image_raw, this is the standard naming scheme.

**Review by irwineffect — CHANGES_REQUESTED** [review]

File: src/vision/image_publisher.cpp (hunk 1 above)
Comment (dcrossman) [reply src/vision/image_publisher.cpp:30]: Ok I renamed them. The _raw part should be added by the image transport.

**Review by irwineffect — APPROVED** [review]

**Pull request merged**